Examples/Framework/include/ActsExamples/Framework/BufferedReader.hpp

```diff
@@ -0,0 +1,73 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Utilities/Logger.hpp"
+#include "ActsExamples/Framework/AlgorithmContext.hpp"
+#include "ActsExamples/Framework/IReader.hpp"
+#include "ActsExamples/Framework/ProcessCode.hpp"
+
+#include <utility>
+
+namespace ActsExamples {
+
+class WhiteBoard;
+
+/// Event data reader that takes a concrete reader instance, reads a number of
+/// events in a buffer, and selects events from that buffer instead of directly
+/// reading them from disk.
+/// The purpose is to avoid IO bottlenecks in timing measurements
+class BufferedReader final : public IReader {
+ public:
+  struct Config {
+    /// The downstream reader that should be used
+    std::shared_ptr<IReader> downstreamReader;
+
+    /// The seed for sampling events from the buffer
+    std::size_t selectionSeed = 123456;
+
+    /// Buffer size. The reader will throw and exception if the downstream
+    /// reader does not provide enough events
+    std::size_t bufferSize;
```

Comment:
This is not guaranteed to be initialized. Should we default this to some (low-ish) number?

Also maybe add a comment that this is cached in memory and therefore has memory footprint implications.

Reply:
good point. I think it should be defaulted to 1.